app/client/lib/uploads.ts

```diff
@@ -123,9 +123,23 @@ export async function uploadFiles(
     }
   }
 
-  return new Promise<UploadResult>((resolve, reject) => {
+  return uploadFormData(docUrl(options.docWorkerUrl, UPLOAD_URL_PATH), formData, onProgress);
+}
+
+/**
+ * POSTs the provided form data to the given endpoint.
+ * Provides progress tracking, error handling and promises.
+ * @param {string} url - Endpoint to send form data to.
+ * @param {FormData} formData - Data to send
+ * @param {ProgressCB} onProgress - Called periodically during the upload
+ * @returns {Promise<any>} - Parsed JSON from the endpoint. Uses `any` as no validation is performed.
+ */
+export async function uploadFormData(
```

Comment:
Is this used anywhere else?

Reply:
Only in 'uploadFiles', which it was extracted from.

It was going to be used elsewhere, but I guess I went a different route.

I'm happy to leave this in though if you are, as it splits up a long function (`uploadFiles`), making it more readable.

Reply:
Fine with leaving, maybe just remove the export, to detect dead code more easily in the future.